File: PlacementCellBackend/Services/Interfaces/ICompanyService.cs
using PlacementCellBackend.Models;

namespace PlacementCellBackend.Services.Interfaces
{
    public interface ICompanyService
    {
        Task<IEnumerable<Company>> GetAllCompaniesAsync();
        Task<Company?> GetCompanyByIdAsync(string id);
        Task<Company> CreateCompanyAsync(Company company);
        Task<bool> UpdateCompanyAsync(string id, Company company);
        Task<bool> DeleteCompanyAsync(string id);
        bool CompanyExists(string id);
    }
}

File: PlacementicsBackend/src/main/java/com/example/placementicsbackend/services/Interfaces/ICompanyService.java
package com.example.placementicsbackend.services.Interfaces;

import com.example.placementicsbackend.models.Company;
import java.util.List;

public interface ICompanyService {

    List<Company> getAllCompanies();

    Company getCompanyById(String id);

    Company createCompany(Company company);

    boolean updateCompany(String id, Company company);

    boolean deleteCompany(String id);

    boolean companyExists(String id);
}
